qutip/visualization.py

```diff
@@ -710,51 +753,72 @@ def matrix_histogram(M, xlabels=None, ylabels=None, title=None, limits=None,
     ypos = ypos.T.flatten() + 0.5
     zpos = np.zeros(n)
     dx = dy = (1 - default_opts['bars_spacing']) * np.ones(n)
-    dz = np.real(M.flatten())
 
-    if isinstance(limits, list) and len(limits) == 2:
+    bar_M = _get_matrix_components(bar_style, M, 'bar_style')
+
+    if isinstance(limits, list) and \
+            len(limits) == 2:
         z_min = limits[0]
         z_max = limits[1]
     else:
-        z_min = min(dz)
-        z_max = max(dz)
+        z_min = min(bar_M)
+        z_max = max(bar_M)
         if z_min == z_max:
             z_min -= 0.1
             z_max += 0.1
 
-    if default_opts['cbar_to_z']:
-        norm = mpl.colors.Normalize(min(dz), max(dz))
+    color_M = _get_matrix_components(color_style, M, 'color_style')
+
+    if isinstance(color_limits, list) and \
+            len(color_limits) == 2:
+        c_min = color_limits[0]
+        c_max = color_limits[1]
     else:
-        norm = mpl.colors.Normalize(z_min, z_max)
-    cmap = _truncate_colormap(default_opts['cmap'],
-                              default_opts['cmap_min'],
-                              default_opts['cmap_max'])
-    colors = cmap(norm(dz))
-
-    if ax is None:
-        fig = plt.figure(figsize=default_opts['figsize'])
-        ax = _axes3D(fig,
-                     azim=default_opts['azim'] % 360,
-                     elev=default_opts['elev'] % 360)
-        ax.set_proj_type(default_opts['proj_type'])
-
-    ax.bar3d(xpos, ypos, zpos, dx, dy, dz, color=colors,
+        if color_style == 'phase':
+            c_min = -pi
+            c_max = pi
+        else:
+            c_min = min(color_M)
+            c_max = max(color_M)
+
+        if c_min == c_max:
+            c_min -= 0.1
+            c_max += 0.1
+
+    norm = mpl.colors.Normalize(c_min, c_max)
+
+    if cmap is None:
+        # change later
+        if color_style == 'phase':
+            cmap = _cyclic_cmap()
+        else:
+            cmap = _sequential_cmap()
+
+    colors = cmap(norm(color_M))
+
+    colors[:, 3] = default_opts['bars_alpha']
+
+    if default_opts['threshold'] is not None:
+        colors[:, 3] = 1 * (bar_M >= default_opts['threshold'])
```

Comment:
This overwrite the `bars_alpha`, you can fix with:
```suggestion
    if default_opts['threshold'] is not None:
        colors[:, 3] *= 1 * (bar_M >= default_opts['threshold'])
```

Reply:
Thanks! I fixed it.